libs/events/import/api/src/lib/import/agenda-basel.service.ts

```diff
@@ -0,0 +1,255 @@
+import {Inject, Injectable} from '@nestjs/common'
+import {
+  CreateEventParams,
+  EventsProvider,
+  ImportedEventParams,
+  ImportedEventsParams
+} from './events-import.service'
+import {ImportedEventsDocument, Event} from './events-import.model'
+import {Cache} from 'cache-manager'
+import moment from 'moment'
+import fetch from 'node-fetch'
+import {htmlToSlate} from 'slate-serializers'
+import xml2js from 'xml2js'
+import axios from 'axios'
+import {ArrayBufferUpload, MediaAdapterService} from '@wepublish/image/api'
+import {XMLEventType} from './xmlTypes'
+import {CACHE_MANAGER} from '@nestjs/cache-manager'
+import {EventStatus, PrismaClient} from '@prisma/client'
+
+const AGENDA_BASEL_URL = 'https://www.agendabasel.ch/xmlexport/kzexport-basel.xml'
+
+const getFallbackDesc = (source: string) => `<p>Event imported from ${source}</p>`
+
+const parseAndCacheData = async (cacheManager: Cache, source: string): Promise<Event[]> => {
+  const parser = new xml2js.Parser()
+  const urlToQuery = AGENDA_BASEL_URL
+
+  async function getXMLfromURL(url: string) {
+    try {
+      const response = await fetch(url)
+      const content = await response.text()
+      const data = await parser.parseStringPromise(content)
+
+      return data
+    } catch (e) {
+      throw Error('Unable to get any data from the provided xml.')
+    }
+  }
+
+  const eventsParsedXML = await getXMLfromURL(urlToQuery)
+  const events = eventsParsedXML['kdz:exportActivities']?.Activities[0]?.Activity
+
+  // only take events that take time in the future
+  const upcomingEvents = events.filter((event: XMLEventType) => upcomingOnly(event))
+
+  // parse their structure to ours
+  const importedEvents = upcomingEvents?.map((a: XMLEventType) => {
+    return parseXMLEventToWpEvent(a, source)
+  })
+
+  // save in cache
+  const ttl = 8 * 60 * 60 * 1000 // 8 hours
+  await cacheManager.set('parsedEvents', importedEvents, ttl)
```

Comment:
Wouldn't it make more sense to let the service decide how to cache? Then we also don't need to pass the cacheManager down.

Reply:
so where would you put this logic?

Reply:
in the service and pass it down